deployment/environment/devenv/chain.go

```
@@ -109,68 +144,158 @@ func (c *ChainConfig) ToRPCs() []deployment.RPC {
-					if err != nil {
-						return blockNumber, fmt.Errorf("failed to get confirmed receipt for chain %s: %w", chainInfo.ChainName, err)
-					}
-					if receipt == nil {
-						return blockNumber, fmt.Errorf("receipt was nil for tx %s chain %s", tx.Hash().Hex(), chainInfo.ChainName)
-					}
-					blockNumber = receipt.BlockNumber.Uint64()
-					if receipt.Status == 0 {
-						errReason, err := deployment.GetErrorReasonFromTx(ec, chainCfg.DeployerKey.From, tx, receipt)
-						if err == nil && errReason != "" {
-							return blockNumber, fmt.Errorf("tx %s reverted,error reason: %s chain %s", tx.Hash().Hex(), errReason, chainInfo.ChainName)
+
+			if chainCfg.ChainType == SolChainType {
+				logger.Info("Creating solana programs tmp dir")
+				programsPath, err := os.MkdirTemp("", "solana-programs")
+				logger.Infof("Solana programs tmp dir at %s", programsPath)
+				if err != nil {
+					return err
+				}
+
+				keyPairDir, err := os.MkdirTemp("", "solana-keypair")
+				logger.Infof("Solana keypair dir at %s", keyPairDir)
+				if err != nil {
+					return err
+				}
+
+				logger.Infof("Deployer newChains %v", chainCfg.DeployerKey)
+				logger.Infof("Deployer newChains %v", chainCfg.DeployerKey)
+				keyPairPath, err := generateSolanaKeypair(chainCfg.SolDeployerKey, keyPairDir)
+				sc := solRpc.New(chainCfg.HTTPRPCs[0].External)
+				solSyncMap.Store(chainDetails.ChainSelector, deployment.SolChain{
+					Selector:    chainDetails.ChainSelector,
+					Client:      sc,
+					DeployerKey: &chainCfg.SolDeployerKey,
+					KeypairPath: keyPairPath,
+					URL:         chainCfg.HTTPRPCs[0].External,
+					WSURL:       chainCfg.WSRPCs[0].External,
+					Confirm: func(instructions []solana.Instruction, opts ...solCommonUtil.TxModifier) error {
+						_, err := solCommonUtil.SendAndConfirm(
+							context.Background(), sc, instructions, chainCfg.SolDeployerKey, solRpc.CommitmentConfirmed, opts...,
+						)
+						if err != nil {
+							panic(err)
```

why panic instead of return?